Sources/IRGen/Expression/IRExpression.swift

```diff
@@ -36,33 +52,41 @@ struct IRExpression {
     case .externalCall:
       fatalError()
     case .identifier(let identifier):
-      return IRIdentifier(identifier: identifier, asLValue: asLValue).rendered(functionContext: functionContext)
+      return GeneratedCode(preamble, IRIdentifier(identifier: identifier, asLValue: asLValue)
+        .rendered(functionContext: functionContext))
     case .variableDeclaration(let variableDeclaration):
-      return IRVariableDeclaration(variableDeclaration: variableDeclaration).rendered(functionContext: functionContext)
+      return GeneratedCode(preamble, IRVariableDeclaration(variableDeclaration: variableDeclaration)
+        .rendered(functionContext: functionContext))
     case .literal(let literal):
-      return IRLiteralToken(literalToken: literal).rendered()
+      return GeneratedCode(preamble, IRLiteralToken(literalToken: literal).rendered())
     case .arrayLiteral(let arrayLiteral):
       for e in arrayLiteral.elements {
         guard case .arrayLiteral(_) = e else {
           fatalError("Cannot render non-empty array literals yet")
         }
       }
-      return "0"
+      return GeneratedCode(preamble, "0")
     case .dictionaryLiteral(let dictionaryLiteral):
       guard dictionaryLiteral.elements.count == 0 else { fatalError("Cannot render non-empty dictionary literals yet") }
-      return "0"
+      return GeneratedCode(preamble, "0")
     case .self(let `self`):
-      return IRSelf(selfToken: self, asLValue: asLValue).rendered(functionContext: functionContext)
-    case .subscriptExpression(let subscriptExpression):
-      return IRSubscriptExpression(subscriptExpression: subscriptExpression, asLValue: asLValue)
+      return IRSelf(selfToken: self, asLValue: asLValue)
         .rendered(functionContext: functionContext)
+    case .subscriptExpression(let subscriptExpression):
+      return IRSubscriptExpression(subscriptExpression: subscriptExpression,
+                                   asLValue: asLValue).rendered(functionContext: functionContext)
     case .sequence(let expressions):
-      return expressions.map {
-            IRExpression(expression: $0, asLValue: asLValue).rendered(functionContext: functionContext)
-        }.joined(separator: "\n")
+      let (p, c) = expressions.reduce(("", ""), {
```

Comment:
Would it not make more sense to reduce over a `GeneratedCode` structure itself? On each iteration you just create a new one.